### DISCUSSION TITLE
How to interpret webhooks events?

### DISCUSSION THREAD

**Discussion by mzagmajster** [postalserver/postal#3062] [Development discussions]
Dear Community,

looking at this docs: https://docs.postalserver.io/developer/webhooks

I am looking to integrate postal with a software and have questions about webhook processing. 

1. How would I detect soft or hard bounce?
2. How would detect FBL complaint?
3. How would I detect unsubscribe action from the webhooks?

Is it possible that these are actually just missing from postal, is there any other way to get this info (with additional API calls)?

Thanks in advance.

**Comment by willpower232** [comment]
1. soft bounces are MessageDelayed, hard bounces are MessageDeliveryFailed

2. this is not a feature of Postal, you would probably have to parse the exact error message for MessageDeliveryFailed

3. this is not a feature of Postal, you have to implement your own subscription list and your own subscribe and unsubscribe pages

**Reply by vasilieva-sasha** [reply]
So in case of soft bounce I will recieve message with status MessageDelayed and then another MessageBounced event?
And will I recieve MessageBounced event in case of hard bounce?

**Reply by willpower232** [reply]
You would only receive a MessageBounced if Postal has received a bounce email. Usually Postal would consider the message as delivered until it received the bounce email from the receiving server.

**Reply by vasilieva-sasha** [reply]
Thank You! 
How can I detect soft bounce from hard bounce in MessageBounced event?

**Reply by willpower232** [reply]
I don't think Postal would be able to identify the difference on its own, you would have to access the bounce message yourself and see what it said.
